### PR TITLE
Refactor Csv reader to pass kwargs to pandas

### PR DESCRIPTION
**Pull request by alejandro-salgado**
- Refactor Csv reader to pass kwargs to pandas

### CODE REVIEW

File: src/allotropy/parsers/lines_reader.py [hunk 1]
```diff
@@ -132,19 +132,15 @@ def pop_csv_block_as_lines(self, empty_pat: str = EMPTY_STR_PATTERN) -> list[str
     def pop_csv_block_as_df(
         self,
         empty_pat: str = EMPTY_STR_PATTERN,
-        *,
-        header: Optional[Union[int, Literal["infer"]]] = None,
-        sep: Optional[str] = ",",
-        as_str: bool = False,
```
Comment (mdlu) [inline src/allotropy/parsers/lines_reader.py:138]: confirming that this functionality is no longer used/needed?

File: src/allotropy/parsers/lines_reader.py [hunk 2]
```diff
@@ -160,19 +156,15 @@ def pop_as_series(self, sep: str = " ") -> Optional["pd.Series[str]"]:
 
     def lines_as_df(
         self,
-        *,
         lines: list[str],
-        header: Optional[Union[int, Literal["infer"]]] = None,
-        sep: Optional[str] = ",",
-        as_str: bool = False,
```
Comment (mdlu) [inline src/allotropy/parsers/lines_reader.py:167]: same as above

**Review by mdlu — APPROVED** [review]

File: src/allotropy/parsers/lines_reader.py (hunk 1 above)
Comment (alejandro-salgado) [reply src/allotropy/parsers/lines_reader.py:138]: yes, this values are no longer needed

File: src/allotropy/parsers/lines_reader.py [hunk 3]
```diff
@@ -132,19 +132,15 @@ def pop_csv_block_as_lines(self, empty_pat: str = EMPTY_STR_PATTERN) -> list[str
     def pop_csv_block_as_df(
         self,
         empty_pat: str = EMPTY_STR_PATTERN,
-        *,
-        header: Optional[Union[int, Literal["infer"]]] = None,
```
[the remark this answers is not in the document]
Comment (alejandro-salgado) [reply src/allotropy/parsers/lines_reader.py:136]: I redefined the header parameter.

**Review by mdlu — APPROVED** [review]

**Review by nathan-stender — APPROVED** [review]

**Pull request merged**